src/runtime/proxy-component.ts

```diff
@@ -62,6 +62,21 @@ export const proxyComponent = (Cstr: d.ComponentConstructor, cmpMeta: d.Componen
       prototype.attributeChangedCallback = function (attrName: string, _oldValue: string, newValue: string) {
         plt.jmp(() => {
           const propName = attrNameToPropName.get(attrName);
+
+          // the attr changed callback runs prior to the connected callback where we have logic that also
+          // has similar logic in it to support lazy properties so if we attempt to unshadow in both places
+          // to cover the case where an attr was set inline on the non-upgrade element and then the property
+          // was programatically set which will be handled here or the case where the attr was not set in
+          // which case the connectedCallback will catch and unshadow.
+          // https://developers.google.com/web/fundamentals/web-components/best-practices#lazy-properties
+          // we should think about whether or not we actually want to be reflecting the attributes to properties
+          // here given that this goes against best practices outlined here
+          // https://developers.google.com/web/fundamentals/web-components/best-practices#avoid-reentrancy
```

Comment:
I took a stab at reworking this comment block a section a bit, LMK what you think (especially if I didn't capture the original messaging correctly 🙃)

```suggestion
          // The attributeChangedCallback may run prior to Stencil's connectedCallback, which may also attempt to
          // unshadow lazy properties. The case where an attribute was set inline on the non-upgraded element:
          // ```html
          //    <!-- this component has _not_ been upgraded yet -->
          //    <my-component some-attribute="some-value></my-component>
          // ```
          //  and the property was subsequently set programmatically shall be handled here.
          //
          //  The case where the attribute was set inline but was not set programmatically shall be handled/unshadowed
          //  by connectedCallback.
          //
          //  https://developers.google.com/web/fundamentals/web-components/best-practices#lazy-properties
          //
          //  TODO(STENCIL-16) we should think about whether or not we actually want to be reflecting the attributes to
          //  properties here given that this goes against best practices outlined here
          //  https://developers.google.com/web/fundamentals/web-components/best-practices#avoid-reentrancy
```